File: algorithms/0345.reverse-vowels-of-a-string/solution.rs
impl Solution {
    pub fn reverse_vowels(s: String) -> String {
        let vowels = vec!['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
        let mut stack: Vec<char> = s.chars().filter(|x| vowels.contains(x)).collect();

        let mut res: Vec<char> = vec![];
        for c in s.chars() {
            if vowels.contains(&c) {
                res.push(stack.pop().unwrap());
            } else {
                res.push(c);
            }
        }

        res.iter().collect()
    }
}

File: algorithms/0345.reverse-vowels-of-a-string/solution.go
func reverseVowels(s string) string {
	vowels := []rune{'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
	stack := []rune{}
	for _, r := range s {
		if contains(vowels, r) {
			stack = append(stack, r)
		}
	}

	res := ""
	for _, r := range s {
		if contains(vowels, r) {
			x := stack[len(stack)-1]
			stack = stack[:len(stack)-1]

			res += string(x)
		} else {
			res += string(r)
		}
	}

	return res
}

func contains(arr []rune, x rune) bool {
	for _, v := range arr {
		if v == x {
			return true
		}
	}
	return false
}
